### PR TITLE
add team goals functionality 

### PR DESCRIPTION
**Pull request by bkchilidawg**
Add functionality of most and least goals scored in a game for the teams

### CODE REVIEW

File: lib/stats.rb [hunk 1]
```diff
@@ -315,5 +315,35 @@ def goal_diffs
...
+
+      if goals_scored > team_highest_goals[team_id]
+        team_highest_goals[team_id] = goals_scored
+      end
+    end
+
+  team_highest_goals
+end
+
+def least_goals_scored_totals
```
Comment (kameronk92) [inline lib/stats.rb:333]: Nice work on these, I wonder if there is a way to condense these into one method

File: spec/stat_tracker_spec.rb [hunk 2]
```diff
@@ -212,5 +212,18 @@
     end
   end
 
+  describe '#most_goals_scored' do
```
Comment (kameronk92) [inline spec/stat_tracker_spec.rb:215]: This looks clean!

File: spec/stats_spec.rb [hunk 3]
```diff
@@ -137,4 +137,16 @@
       expect(@stat_tracker.seasonal_summaries["6"]["20122013"][:postseason]).to eq(expected)
     end
   end
+
+  describe '#most_goals_scored_totals' do
+    it 'returns the highest number of goals scored by any team in a single game' do
+      expect(@stat_tracker.most_goals_scored_totals).to eq({"3"=>2, "4"=>4, "6"=>3})
```
Comment (kameronk92) [inline spec/stats_spec.rb:143]: nice organization here.

Comment (bkchilidawg) [reply spec/stats_spec.rb]: thanks!

File: lib/stats.rb [hunk 4]
```diff
@@ -315,5 +316,24 @@ def goal_diffs
     goal_diffs
   end
 
-  ##== TEAM HELPERS ==##
+  def max_min_goals
```
Comment (kameronk92) [inline lib/stats.rb:319]: great work with the refacto on this

File: lib/stats.rb [hunk 5]
```diff
@@ -278,7 +278,8 @@ def average_goals_against(season_type, season_id, team_id)
   def teams_hash
     if @teams_hash.nil?
       @teams_hash = {}
-
+      
+      @teams_hash[:max_min_goals] = max_min_goals
```
Comment (kameronk92) [inline lib/stats.rb:282]: great use of the teams hash

**Review by kameronk92 — APPROVED** [review]

**Comment by kameronk92** [comment]
Nice job with the refactor

**Pull request merged**